src/sentry/api/endpoints/rule_snooze.py

```diff
@@ -0,0 +1,96 @@
+import datetime
+
+from rest_framework import serializers, status
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry import audit_log, features
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.project import ProjectAlertRulePermission, ProjectEndpoint
+from sentry.api.exceptions import ConflictError
+from sentry.api.serializers import serialize
+from sentry.api.serializers.rest_framework.base import CamelSnakeSerializer
+from sentry.incidents.models import AlertRule
+from sentry.models import Rule, RuleSnooze
+
+
+class RuleSnoozeSerializer(CamelSnakeSerializer):
+    user_id = serializers.IntegerField(required=False, allow_null=True)
+    rule = serializers.BooleanField(required=False, allow_null=True)
+    alert_rule = serializers.BooleanField(required=False, allow_null=True)
+    until = serializers.DateTimeField(required=False, allow_null=True)
+
+
+@region_silo_endpoint
+class RuleSnoozeEndpoint(ProjectEndpoint):
+    permission_classes = (ProjectAlertRulePermission,)
+
+    def post(self, request: Request, project, rule_id) -> Response:
+        if not features.has("organizations:mute-alerts", project.organization, actor=None):
+            return Response(
+                {"detail": "This feature is not available for this organization."}, status=401
+            )
+
+        serializer = RuleSnoozeSerializer(data=request.data)
+        if serializer.is_valid():
+            data = serializer.validated_data
+
+            issue_alert_rule = None
+            metric_alert_rule = None
+
+            if data.get("rule"):
+                try:
+                    issue_alert_rule = Rule.objects.get(id=rule_id)
+                except Rule.DoesNotExist:
+                    raise serializers.ValidationError("Rule does not exist")
+
+            if data.get("alert_rule"):
+                try:
+                    metric_alert_rule = AlertRule.objects.get(id=rule_id)
+                except AlertRule.DoesNotExist:
+                    raise serializers.ValidationError("Rule does not exist")
+
+            # don't allow editing of a rulesnooze object for a given rule and user (or no user)
+            if RuleSnooze.objects.filter(
+                user_id=data.get("user_id"), rule=issue_alert_rule, alert_rule=metric_alert_rule
+            ).exists():
+                raise ConflictError("RuleSnooze already exists for this rule and scope.")
+
+            RuleSnooze.objects.create(
```

Comment:
how do we prevent people from creating rulesnooze's for organizations / projects they don't belong to? from what I see, there's no project_id in this model, so couldn't I pass in an arbitrary alert rule in theory?